src/essentialsTP/essentialsTP.php

```diff
@@ -5,1373 +5,1191 @@
-                $this->world = $player->getLevel()->getName();
-                $this->home_loc = "bed";
-                $this->prepare = $this->db2->prepare("SELECT player,title,x,y,z,world FROM homes WHERE player = :name AND title = :title");
-                $this->prepare->bindValue(":name", $this->username, SQLITE3_TEXT);
-                $this->prepare->bindValue(":title", $this->home_loc, SQLITE3_TEXT);
-                $this->result = $this->prepare->execute();
-                $sql          = $this->fetchall();
-                if( count($sql) > 0 )
-                {
-                    $this->prepare = $this->db2->prepare("UPDATE homes SET world = :world, title = :title, x = :x, y = :y, z = :z WHERE player = :name AND title = :title");
-                    $this->prepare->bindValue(":name", $this->username, SQLITE3_TEXT);
-                    $this->prepare->bindValue(":title", $this->home_loc, SQLITE3_TEXT);
-                    $this->prepare->bindValue(":world", $this->world, SQLITE3_TEXT);
-                    $this->prepare->bindValue(":x", $this->player_cords['x'], SQLITE3_TEXT);
-                    $this->prepare->bindValue(":y", $this->player_cords['y'], SQLITE3_TEXT);
-                    $this->prepare->bindValue(":z", $this->player_cords['z'], SQLITE3_TEXT);
-                    $this->result = $this->prepare->execute();
-
-                }
-                else
-                {
-                    $this->prepare = $this->db2->prepare("INSERT INTO homes (player, title, world, x, y, z) VALUES (:name, :title, :world, :x, :y, :z)");
-                    $this->prepare->bindValue(":name", $this->username, SQLITE3_TEXT);
-                    $this->prepare->bindValue(":title", $this->home_loc, SQLITE3_TEXT);
-                    $this->prepare->bindValue(":world", $this->world, SQLITE3_TEXT);
-                    $this->prepare->bindValue(":x", $this->player_cords['x'], SQLITE3_TEXT);
-                    $this->prepare->bindValue(":y", $this->player_cords['y'], SQLITE3_TEXT);
-                    $this->prepare->bindValue(":z", $this->player_cords['z'], SQLITE3_TEXT);
-                    $this->result = $this->prepare->execute();
-
-                }
-            }
-        }
-    }
-
-    public function onBlockTap(PlayerInteractEvent $event)
-    {
-		if ($event->isCancelled()) return true;
+class essentialsTP extends PluginBase implements Listener
+{
+
+	/** @var \SQLite3 */
+	private $db2;
+	/** @var string */
+	public $username;
+	/** @var string */
+	public $world;
+	/** @var string */
+	public $home_loc;
+	/** @var string */
+	public $warp_loc;
+	/** @var Position[] */
+	public $death_loc;
+	/** @var Config */
+	public $config;
+	/** @var int[] */
+	public $player_cords;
+	/** @var string */
+	public $tp_sender;
+	/** @var string */
+	public $tp_reciver;
+	/** @var \SQLite3Result */
+	public $result;
+	/** @var \SQLite3Stmt */
+	public $prepare;
+	private $tpa_cooldown;
+
+	public function fetchall()
```

Comment:
preserve the original format, don't change for no reason. it is making the PR harder to review.